### PR TITLE
use enginetools/walk

### PR DESCRIPTION
**Pull request by tylermann**
This updates #1 to use the walk command from enginetools which handles checking the `/engine_config.json` file's `include_paths` and `exclude_paths` and filtering on the patterns in a standardized way across all engines. Easiest way to convert this for this engine was to just use the cmd flag: `-cmd=/go/bin/govet-engine` which calls `govet-engine` for each matched path. Other alternative for engines that have a slower startup time is to not use the `-cmd` flag and pipe the matched paths into the stdinput as is done in the eslint-engine due to the slow startup time.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,12 +1,14 @@
 # Need golang 1.11 because 1.12 deprecates "go tool vet".
 # See: https://github.com/flycheck/flycheck/issues/1523
-FROM golang:1.11-alpine
+FROM golang:1.11-alpine as build
 
-WORKDIR /go/src/app
+COPY --from=439412406194.dkr.ecr.us-east-1.amazonaws.com/pullrequest/enginetools /app/build/walk /enginetools/walk
```
Comment (mcurtiss) [inline Dockerfile]: I don't know if it's ideal to have the full URL of our ECR registry in the Dockerfile. (As in: I honestly don't know).  Would it be cleaner to use an environment variable somehow?

File: cmd/govet-engine/main.go [hunk 2]
```diff
@@ -35,26 +34,18 @@ func GetEngineConfig() (engineconfig, error) {
 }
 
 func main() {
+	if len(os.Args) < 2 {
+		log.Fatal("missing path parameter")
```
Comment (mcurtiss) [inline cmd/govet-engine/main.go]: We should document this a little better. Perhaps a comment at the top of the file?

    `Usage: $ govet-engine <filename>`

?

File: cmd/govet-engine/main.go [hunk 3]
```diff
@@ -35,26 +34,18 @@ func GetEngineConfig() (engineconfig, error) {
...
+	}
+	path := os.Args[1]
+
 	_, err := GetEngineConfig() // TODO: use the config!
```
Comment (mcurtiss) [inline cmd/govet-engine/main.go]: Do we still expect to use engine configs, or does enginetools eliminate the need for this?

If so, we should document how to use (is it a flag or is it os.Args[2]) ?

File: Dockerfile (hunk 1 above)
Comment (tylermann) [reply Dockerfile]: yeah we could eventually set up the Makefile's `make build` to auto include the env variable for this prefix or something. I think for now it is fine and we can update them all later if needed. I'm still trying to figure out the best way to work with these ecr hosted images but it seems like there isn't a way to override even locally without naming it the full name.

File: cmd/govet-engine/main.go (hunk 2 above)
Comment (tylermann) [reply cmd/govet-engine/main.go]: yeah that's a good idea

File: cmd/govet-engine/main.go (hunk 3 above)
Comment (tylermann) [reply cmd/govet-engine/main.go]: you may still need it, it depends on if your engine has more configuration options or not. the enginetools/walk handles the `include_paths` and `exclude_paths` only for now.

**Pull request merged**